### PR TITLE
Making Table of contents more usable.

### PR DESCRIPTION
**Pull request by anjulgarg**
# What
Table of contents is more stylized now. Looks sharper and has consistent fonts.
TOC has 
- reduced font sized 
- tighter margins
- Margins between list items.
- Margins between anchors.
- responsive. Will be hidden if the screen is not wide enough (1200px).

If viewport is wider than 1200px
image

If viewport width is between 900px and 1200px
image

If viewport is smaller than 900px
image

# Why
For narrow screens, TOC would overlap content making it impossible to read.

### CODE REVIEW

File: packages/storybook/.storybook/TOC.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { mergeStyles } from '@fluentui/react';
+import React, { useState } from 'react';
+import { BackToTop, TableOfContents } from 'storybook-docs-toc';
+
+export const TOC = (props: {children: React.ReactNode}): JSX.Element => {
```
Comment (ytyeason) [inline packages/storybook/.storybook/TOC.tsx:5]: can we add a comment here explaining the meaning of TOC?

**Review by ytyeason — APPROVED** [review]

File: packages/storybook/.storybook/TOC.tsx (hunk 1 above)
Comment (anjulgarg) [reply packages/storybook/.storybook/TOC.tsx:5]: Sure

File: packages/storybook/.storybook/TOC.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+/**
+ * Table Of Contents component used for displaying a floating table of content
+ * inside Storybook Docs.
+ * @param props 
+ * @returns 
+ */
+export const TOC = (props: {children: React.ReactNode}): JSX.Element => {
+
+  const [narrowDocs, setnarrowDocs] = useState(false);
```
Comment (alcail) [inline packages/storybook/.storybook/TOC.tsx:13]: nit: set**N**arrowDocs

**Review by alcail — APPROVED** [review]

File: packages/storybook/.storybook/TOC.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  const [hideToc, setHideToc] = useState(false);
+
+  window.matchMedia('(max-width: 1200px)').addEventListener('change', (e) => {
```
Comment (allenplusplus) [inline packages/storybook/.storybook/TOC.tsx:16]: Could we put these inside a useEffect and when the component unmounts unregister the listeners?

**Review by allenplusplus — APPROVED** [review]

File: packages/storybook/.storybook/TOC.tsx (hunk 3 above)
Comment (anjulgarg) [reply packages/storybook/.storybook/TOC.tsx]: Yes. Thanks for pointing it out.

**Pull request merged**